File: VB/第二套/11900-1060307/Services/DataService.vb
Imports System.IO

Namespace Services
    Public Interface IDataService
        Function LoadRandomNumbers(path As String) As IEnumerable(Of Double)
    End Interface

    Public Class DataService
        Implements IDataService

        Public Function LoadRandomNumbers(path As String) As IEnumerable(Of Double) Implements IDataService.LoadRandomNumbers
            Dim numbers As New List(Of Double)()
            If File.Exists(path) Then
                For Each line In File.ReadAllLines(path)
                    Dim val As Double
                    If Double.TryParse(line, val) Then
                        numbers.Add(val)
                    End If
                Next
            End If
            Return numbers
        End Function
    End Class
End Namespace

File: C#/第二套/11900-1060307/Services/DataService.cs
namespace PokerGame.Services;

using System.IO;

public interface IDataService
{
    Task<(int rounds, List<double> randomNumbers)> ReadDataAsync(string filePath);
}

public class DataService : IDataService
{
    public async Task<(int rounds, List<double> randomNumbers)> ReadDataAsync(string filePath)
    {
        if (!File.Exists(filePath)) return (0, new List<double>());

        var lines = await File.ReadAllLinesAsync(filePath);
        if (lines.Length == 0) return (0, new List<double>());

        if (!int.TryParse(lines[0], out int rounds)) rounds = 0;

        var randomNumbers = lines.Skip(1)
            .Select(line => double.TryParse(line, out double val) ? val : -1.0)
            .Where(val => val >= 0)
            .ToList();

        return (rounds, randomNumbers);
    }
}
